### PR TITLE
Fixes multiple home pages

### LINKED ISSUE #10657: Multiple home pages are not working
**Describe the issue you encountered:**

Automated test is failing for this feature. https://travis-ci.org/brave/browser-laptop/jobs/268215305#L3828

```
General Panel homepage multiple homepages direct input:
     Promise was rejected with the following reason: Promise was rejected with the following reason: timeout
  Error
General Panel homepage multiple from scratch:
     Promise was rejected with the following reason: timeout
```

### PR DESCRIPTION
**Pull request by NejcZdovc**
Submitter Checklist:

- [x] Submitted a [ticket](https://github.com/brave/browser-laptop/issues) for my issue if one did not already exist.
- [x] Used Github [auto-closing keywords](https://help.github.com/articles/closing-issues-via-commit-messages/) in the commit message.
- [x] Added/updated tests for this change (for new code or code which already has tests).
- [x] Ran `git rebase -i` to squash commits (if needed).
- [x] Tagged reviewers and labelled the pull request [as needed](https://github.com/brave/browser-laptop/wiki/Pull-request-process).

Resolves #10657

Auditors:

Test Plan:
- covered by automated tests

Reviewer Checklist:

Tests

- [ ] Adequate test coverage exists to prevent regressions
- [ ] Tests should be independent and work correctly when run individually or as a suite [ref](https://github.com/brave/browser-laptop/wiki/Code-Guidelines#test-dependencies)
- [ ] New files have MPL2 license header

### CODE REVIEW

File: js/stores/appStore.js [hunk 1]
```diff
@@ -129,14 +129,32 @@ function handleChangeSettingAction (settingKey, settingValue) {
...
+        BrowserWindow.getAllWindows().forEach(function (wnd) {
+          wnd.webContents.setZoomLevel(newZoomLevel)
+        })
+        break
+      }
+    case settings.HOMEPAGE:
+      {
+        let homeArray = settingValue.split('|')
+        homeArray = homeArray.map(page => {
+          page = page && page.trim()
```
Comment (darkdh) [inline js/stores/appStore.js]: any reasons for why not just using `page = page.trim()`?

Comment (NejcZdovc) [reply js/stores/appStore.js]: yeah I can remove this one, it was copied from another place

**Comment by NejcZdovc** [comment]
@darkdh I would keep this scenario `https://brave.com|||`, because you could use this to have one page and three empty tabs. Not sure if this is an actual use case, but this will not produce any errors.

File: js/stores/appStore.js (hunk 1 above)
Comment (NejcZdovc) [reply js/stores/appStore.js]: Done

File: js/about/preferences.js [hunk 2]
```diff
@@ -137,17 +136,7 @@ class GeneralTab extends ImmutableComponent {
...
-      const punycodeUrl = UrlUtil.getPunycodeUrl(homepageValue)
-      if (punycodeUrl.replace(/\/$/, '') !== homepageValue) {
-        homepageValue = UrlUtil.getPunycodeUrl(homepageValue)
-      }
-
-      // we use | as a separator for multiple home pages
-      homepageValue = homepageValue.replace(/%7C/g, '|')
-    }
-    const homepage = homepageValue && homepageValue.trim()
+    const homepage = getSetting(settings.HOMEPAGE, this.props.settings)
```
Comment (bsclifton) [inline js/about/preferences.js]: I think removing the punycode encoding may cause a regression

Can you please test this with an international domain name?
http://www.i18nguy.com/markup/idna-examples.html

Comment (NejcZdovc) [reply js/about/preferences.js]: Punycode was just moved from component into appState. Please look at line 142 in appState

Comment (NejcZdovc) [reply js/about/preferences.js]: https://github.com/brave/browser-laptop/pull/10659/files#diff-23ca389e2bcb77191b5a9c10900eb3a3R142

Comment (bsclifton) [reply js/about/preferences.js]: ah- totally missed that 😄  perfect!

**Pull request merged**